src/server/src/main/java/io/cassandrareaper/jmx/ClusterFacade.java

```diff
@@ -339,6 +354,40 @@ public String getCassandraVersion(Cluster cluster, Collection<String> endpoints)
     return Lists.newArrayList();
   }
 
+  /**
+   * Get the ranges for the local node (only for sidecar mode).
+   *
+   * @param cluster the cluster to connect to
+   * @param keyspace the keyspace we're getting the ranges for.
+   * @return the list of local token ranges
+   * @throws ReaperException any runtime exception we catch in the process
+   */
+  public List<RingRange> getRangesForLocalEndpoint(Cluster cluster, String keyspace) throws ReaperException {
+    Preconditions.checkArgument(context.config.isInSidecarMode(), "This method is only allowed in sidecar mode");
+    List<RingRange> localRanges = Lists.newArrayList();
+    try {
+      Map<List<String>, List<String>> ranges = getRangeToEndpointMap(cluster, keyspace);
+      JmxProxy jmxProxy = connectAndAllowSidecar(cluster, Arrays.asList(LOCALHOST));
+      String localEndpoint = jmxProxy.getLocalEndpoint();
+      // Filtering ranges for which the local node is a replica
+      // For local mode
+      ranges
+          .entrySet()
+          .stream()
+          .forEach(entry -> {
+            if (entry.getValue().contains(localEndpoint)) {
+              localRanges.add(
+                  new RingRange(new BigInteger(entry.getKey().get(0)), new BigInteger(entry.getKey().get(1))));
+            }
+          });
+
+      return localRanges;
+    } catch (RuntimeException e) {
```

Comment:
what unchecked exceptions are we catching here?

Reply:
right, I've removed the try/catch block.

Reply:
did you forget to push that change?